ui-v2/src/hooks/task-run-concurrency-limits.ts

```diff
@@ -204,3 +211,109 @@ export const useResetTaskRunConcurrencyLimitTag = () => {
 		...rest,
 	};
 };
+
+/**
+ *
+ * @param id
+ * @returns query options for a task-run concurrency limit with active run details that includes details on task run, flow run, and flow
+ */
+export const buildConcurrenyLimitDetailsActiveRunsQuery = (id: string) =>
```

Comment:
Looks like this is hitting our long-standing challenge of n+1 requests to the API. This handles the challenge well, but I'm a little concerned that all these calls are contained in one query, which is then used to suspend rendering until they've all been resolved.

There are a couple of things I'd like to explore:
1. Breaking these calls into separate queries and only suspending rendering for critical data (like the concurrency limit info). We can also investigate using [deferred](https://tanstack.com/router/latest/docs/framework/react/guide/deferred-data-loading) loading with `@tanstack/router` to continue using data loaders without blocking rendering.
2. Update the task run routes to include the flow run name to avoid needing to make another call for the flow runs filter route. I don't know how feasible this is since we'll need to take DB performance into account.

I'll do some exploration in the deployments list, but let's make sure we compare notes because we'll want a consistent approach to this in the application!

Reply:
Sounds good. I can do some more exploration as well.

I opted for this root cause the prefetching and loading mechanism comes out of the box. This way seems to give a faster UX because the data is loaded upon hovering over the Links

Reply:
I’ll try using the selector option in the query.

that way we keep the same query key, but able to call the n+1 APIs for the specific data

Reply:
I think this video is a good explanation on the different options to sync data to the client: https://youtu.be/AuHqwQsf64o?t=566

Just need to determine what category of data critical we need

<img width="1040" alt="Screenshot 2024-12-27 at 2 55 28 PM" src="https://github.com/user-attachments/assets/d630accd-6d5f-463b-a6fb-f80be370decb" />

Reply:
Latest push uses the await/defer approach. IMO, the UX feels the same as just sticking it in a useSuspenseQuery